examples/kernel_ridge_regression.jl

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+# # Kernel Ridge Regression
+#
+# Building on linear regression, we can fit non-linear data sets by introducing a feature space. In a higher-dimensional feature space, we can overfit the data; ridge regression introduces regularization to avoid this. In this notebook we show how we can use KernelFunctions.jl for *kernel* ridge regression.
+
+## Loading and setup of required packages
+using KernelFunctions
+using LinearAlgebra
+using Distributions
+
+## Plotting
+using Plots;
+default(; lw=2.0, legendfontsize=15.0);
+
+using Random: seed!
+seed!(42);
+
+# ## From linear regression to ridge regression
+# Here we use a one-dimensional toy problem. We generate data using the fourth-order polynomial $f(x) = (x+4)(x+1)(x-1)(x-3)$:
+
+f_truth(x) = (x + 4) * (x + 1) * (x - 1) * (x - 3)
+
+x_train = collect(-5:0.5:5)
+x_test = collect(-5:0.1:5)
+
+noise = rand(Uniform(-10, 10), size(x_train))
+y_train = f_truth.(x_train) + noise
+y_test = f_truth.(x_test)
+
+plot(x_test, y_test; label=raw"$f(x)$")
+scatter!(x_train, y_train; label="observations")
+
+# For training inputs $\mathrm{X}=(\mathbf{x}_n)_{n=1}^N$ and observations $\mathbf{y}=(y_n)_{n=1}^N$, the linear regression weights $\mathbf{w}$ using the least-squares estimator are given by
+# $$
+# \mathbf{w} = (\mathrm{X}^\top \mathrm{X})^{-1} \mathrm{X}^\top \mathbf{y}
+# $$
+# We predict at test inputs $\mathbf{x}_*$ using
+# $$
+# \hat{y}_* = \mathbf{x}_*^\top \mathbf{w}
+# $$
+# This is implemented by `linear_regression`:
+
+function linear_regression(X::AbstractVecOrMat, y::AbstractVector, Xstar::AbstractVecOrMat)
```

Comment:
Since we don't dispatch on the arguments anywhere the types are not helpful:
```suggestion
function linear_regression(X, y, Xstar)
```

Reply:
you don't think it'd be helpful for the reader to know what types it should be ?

Reply:
Personally, I don't think `AbstractVecOrMat` is particularly helpful since it does not even provide guarantees about the shape and dimensions. In general, I would put more focus on good explanations, consistent and descriptive variable names, and comments in the code if needed.